### DISCUSSION TITLE
Can aqualinkd work without aqualink rs

### DISCUSSION THREAD

**Discussion by danfairhurst** [aqualinkd/AqualinkD#261] [Q&A] [answered]
I currently have an ESP8266 running esphome and controlling my jandy pump via rs485 directly.
I’m considering getting the Jandy lights that have full aqualink control but I’m wondering if I would need just the light controller and to control them over rs485 (with aqualinkd?) or if I would need the aqualink rs controller.

Can aqualinkd interface with the devices directly over rs485 or does it only interact with aqualink rs?

**Comment by sfeakes** [comment]
You will need an aqualink RS control panel, for AqualinkD to control anything.
I have not come across any lights that are actually RS485 controlled, (not sayin they don't exist). All the ones I have come across are actually controlled by the control panel pulsing power to the lights, and the number of pulses will equate to different colors or dim levels.  So for you setup if you do not have a jandy aqualink RS control panel, it would probably be easiest to simply wire a relay up to your ESP8266 and write some software to pulse the relay contacts. Obviously use the relay to power the lights.

**Reply by danfairhurst** [reply]
I already had control of lights like that via a smart switch, but they broke, so was considering the jandy infinite watercolors which can go any color rather than just presets via cycling, and have their own controller which I believe then connects to aqualink rs via rs485.

I was wondering if aqualinkd would interface with that controller directly and mean I wouldn’t need the rs as well

**Reply by danfairhurst** [reply]
It’s this one: https://www.vitafilters.com/products/jandy-iwclc300w-infinite-watercolors-light-controller-transformer-300w

**Comment by sfeakes** [comment, marked as answer]
AqualinkD will not connect / control that light without a RS control panel.   I also have not seen any decoded protocol for that style light, so to control it directly, you'll probably have to reverse engineer the protocol, and to do that you'll also need a RS control panel.

**Reply by danfairhurst** [reply]
Thanks!

**Reply by robertw** [reply]
Good day Dan I just got a jandy pump and would loove if you could share your esphome code to control it many thanks
